### PR TITLE
Persist tab state without adding to browser history

### PR DESCRIPTION
**Pull request by jwforres**

### CODE REVIEW

**Comment by jwforres** [comment]
@spadgett WDYT of this change?  I can't think of a reason that reloadOnSearch matters on the pages where we have tabs or filters.

**Comment by spadgett** [comment]
+1 `reloadOnSearch: false` seems like exactly what we want

**Comment by benjaminapetersen** [comment]
@jwforres, i think `uib-tab` supports `select=` as a callback, so something like:
```html
select="selectTab('environment')"
```
and the JS could just be:
```javascript
$scope.selectTab = function(what) {
      $location.replace().search('tab', what);
};
```
might be nice, less code, no additional $watch?

**Comment by jwforres** [comment]
@benjaminapetersen in the version we have its on each tab, it adds it to every tab in the view

**Comment by jwforres** [comment]
I agree its less code for the controller, but it doesn't feel right to me to make each tab in the view responsible for making sure the callback gets fired.  In my ideal world the uib-tabset element should have had a callback option for when any of its tabs are selected

**Comment by benjaminapetersen** [comment]
@jwforres, yup, each would get `<uib-tab select="onSelect('tab-name')"`

**Comment by benjaminapetersen** [comment]
Hmm. I think if our views are declarative & we get all the work into a service its a win, but I'm good if you prefer the current approach!

**Comment by jwforres** [comment]
@spadgett can be the deciding vote

**Comment by benjaminapetersen** [comment]
Cool cool!  I'll readily admit I'm an awful controller troll/nag :smile:

**Comment by jwforres** [comment]
@benjaminapetersen how do you feel about the latest version

File: app/scripts/directives/util.js [hunk 1]
```diff
@@ -178,4 +178,24 @@ angular.module('openshiftConsole')
         }
       });
     };
+  })
+  .directive('persistTabState', function($routeParams, $location) {
```
Comment (spadgett) [inline app/scripts/directives/util.js]: Probably worth a comment saying `reloadOnSearch` should be false for the route to use this directive.

**Comment by benjaminapetersen** [comment]
Oooo nice, even less in the controllers!  +1
Only thought, perhaps put it on the tabset itself for context:
```html
<uib-tabset class="mar-top-none" persist-tab-state>
```
Either way though, def looks good to me.

**Comment by jwforres** [comment]
[merge]

**Pull request merged**